server/util/grpc_client/grpc_client.go

```diff
@@ -6,16 +6,28 @@ import (
 	"net/url"
 	"time"
 
+	"github.com/buildbuddy-io/buildbuddy/server/environment"
 	"github.com/buildbuddy-io/buildbuddy/server/rpc/interceptors"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials/google"
 	"google.golang.org/grpc/keepalive"
 )
 
-// DialTarget handles some of the logic around detecting the correct GRPC
+// DialSimple handles some of the logic around detecting the correct GRPC
 // connection type and applying relevant options when connecting.
-func DialTarget(target string, extraOptions ...grpc.DialOption) (*grpc.ClientConn, error) {
-	dialOptions := CommonGRPCClientOptions()
+//
+// NOTE: Clients within BuildBuddy servers (i.e. executor) should use the DialInterservice
+// variant with an env argument.
+func DialSimple(target string, extraOptions ...grpc.DialOption) (*grpc.ClientConn, error) {
```

Comment:
Is there a strong reason to rename DialTarget to DialSimple?

I think DialTarget is clearer and maybe doesn't need to change -- we could just introduce the DialInterservice and use that where necessary?

That should reduce the number of changes in this CL significantly

Reply:
I wanted people to think twice when dialing to make sure they are using the appropriate function. The change is intentionally disruptive so that people don't keep using DialTarget for everything out of momentum.